src/Console/Commands/ImportData.php

```diff
@@ -48,61 +79,63 @@ public function handle(): void
         if (4 != $ipv and 6 != $ipv) {
             throw new Exception('--ipv must be 4 or 6');
         }
-        if ($ipv == 4) {
+        if (4 == $ipv) {
             $model = SubnetV4::class;
         } else {
             $model = SubnetV6::class;
         }
-    
+
         $fd = new SplFileObject($file, 'r');
         while (!$fd->eof()) {
             $row = $fd->fgetcsv();
             if (!$row) {
                 continue;
             }
+
             $row = $this->parseCsv($row);
-            if ('-' == $row['countryCode']) {
+            if ((isset($row['countryCode']) and '-' == $row['countryCode']) or (isset($row['asn']) and '-' == $row['asn'])) {
                 continue;
             }
-            $this->save($model, $row);
+
+            if ($this->option('asn')) {
+                $this->saveAsn($model, $row);
+            } else {
+                $this->save($model, $row);
+            }
         }
     }
 
     /**
      * @param array<string> $data
-     * @return CsvData
+     *
+     * @return IPCsvData|ASNCsvData
      */
     protected function parseCsv(array $data)
     {
-        $columns = [
-            'network_start',
-            'network_end',
-            'countryCode',
-            'country',
-            'region',
-            'city',
-            'latitude',
-            'longitude',
-            'zipcode',
-            'timezone',
-        ];
+        $columns = $this->option('asn') ?
+            $this->asnColumns :
+            $this->ip2LocationColumns;
+
         $count = count($data);
         if ($count != count($columns)) {
             throw new Exception('Csv data is not supported');
```

Comment:
I execute this command and after some time this exception thrown.

![image](https://user-images.githubusercontent.com/16887332/213139064-f1c0b032-6fe3-4a24-90b3-d136bad2b659.png)